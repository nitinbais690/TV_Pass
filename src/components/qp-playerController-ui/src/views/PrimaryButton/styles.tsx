import { StyleSheet } from 'react-native';
import { appDimensionValues, appFontStyle } from 'core/styles/AppStyles';
import { scale } from 'qp-common-ui';

export const styles = (appColors: any) => {
    return StyleSheet.create({
        buttonStyle: {
            elevation: appDimensionValues.xxxxs,
            borderRadius: appDimensionValues.commonButtonHeight / 2,
        },
        containerStyle: {
            flex: 1,
            height: appDimensionValues.commonButtonHeight,
        },
        titleStyle: {
            color: appColors.secondary,
            ...appFontStyle.buttonText,
        },
        unfocused: {
            backgroundColor: appColors.primaryVariant3,
            elevation: 0,
        },
        buttonIconStyle: {
            marginEnd: scale(10),
        },
    });
};
