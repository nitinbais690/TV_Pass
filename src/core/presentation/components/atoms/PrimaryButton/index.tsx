import React, { useState } from 'react';
import { Platform } from 'react-native';
import { Button as RNEButton, ButtonProps as RNEButtonProps } from 'react-native-elements';
import { arrowRightIconHeight, arrowRightIconWidth, styles } from './styles';
import LinearGradient from 'react-native-linear-gradient';
import { appFontStyle } from 'core/styles/AppStyles';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import ArrowRightIcon from 'assets/images/arrow_right_icon.svg';

interface ButtonProps extends RNEButtonProps {
    hasTVPreferredFocus?: boolean;
    primaryButtonIcon?: any;
    primaryTestID?: string;
    primaryAccessibilityLabel?: string;
}

const PrimaryButton = (props: ButtonProps) => {
    const [focussed, setFocussed] = useState(false);
    let buttonStyles = styles(useAppColors());
    let fontStyles = appFontStyle;

    return (
        <RNEButton
            containerStyle={[buttonStyles.containerStyle, props.containerStyle]}
            buttonStyle={[
                buttonStyles.buttonStyle,
                Platform.isTV && focussed ? buttonStyles.focusStyle : buttonStyles.unfocused,
                props.buttonStyle,
            ]}
            titleStyle={props.titleStyle || [fontStyles.body3, buttonStyles.titleStyle]}
            {...props}
            onFocus={() => setFocussed(true)}
            onBlur={() => setFocussed(false)}
            icon={
                Platform.isTV ? (
                    <ArrowRightIcon
                        width={arrowRightIconWidth}
                        height={arrowRightIconHeight}
                        style={Platform.isTV && buttonStyles.iconStyle}
                    />
                ) : (
                    props.primaryButtonIcon && <props.primaryButtonIcon style={buttonStyles.buttonIconStyle} />
                )
            }
            iconRight={Platform.isTV ? true : false}
            hasTVPreferredFocus={props.hasTVPreferredFocus === false ? props.hasTVPreferredFocus : true}
            ViewComponent={LinearGradient} // Don't forget this!
            linearGradientProps={{
                colors: [useAppColors().brandTintLight, useAppColors().brandTintDark],
            }}
            onPress={e => {
                if (props.onPress) {
                    // see here for more details: https://github.com/facebook/react-native/issues/25979
                    props.onPress(e);
                }
            }}
            testID={props.primaryTestID}
            accessibilityLabel={props.primaryAccessibilityLabel}
        />
    );
};

export default PrimaryButton;
