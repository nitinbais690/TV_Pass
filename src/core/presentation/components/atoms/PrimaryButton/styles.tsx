import { StyleSheet, Platform } from 'react-native';
import { appDimensionValues, appFontStyle } from 'core/styles/AppStyles';
import { scale, selectDeviceType } from 'qp-common-ui';

export const arrowRightIconWidth = selectDeviceType({ Tv: scale(16, 0) }, scale(16, 0));
export const arrowRightIconHeight = selectDeviceType({ Tv: scale(10, 0) }, scale(10, 0));

export const styles = (appColors: any) => {
    return StyleSheet.create({
        buttonStyle: {
            elevation: appDimensionValues.xxxxs,
            borderRadius: Platform.isTV ? appDimensionValues.xxxxxs : appDimensionValues.commonButtonHeight / 2,
            justifyContent: Platform.isTV ? 'flex-start' : 'center',
            paddingHorizontal: selectDeviceType({ Tv: scale(10, 0) }, scale(8, 0)),
        },
        containerStyle: {
            flex: 1,
            height: appDimensionValues.commonButtonHeight,
        },
        titleStyle: {
            color: appColors.secondary,
            ...appFontStyle.buttonText,
        },
        unfocused: {
            backgroundColor: appColors.primaryVariant3,
            elevation: 0,
        },
        buttonIconStyle: {
            marginEnd: scale(10),
        },
        iconStyle: {
            position: 'absolute',
            right: 14,
        },
        focusStyle: {
            borderColor: '#ffffff',
            borderWidth: 2,
        },
    });
};
