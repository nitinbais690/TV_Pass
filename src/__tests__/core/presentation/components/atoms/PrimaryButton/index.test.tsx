import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import PrimaryButton from 'core/presentation/components/atoms/PrimaryButton';

jest.mock('core/presentation/hooks/use-app-colors');

describe('primaryButton', () => {
    test('renders correctly', () => {
        const { container } = render(<PrimaryButton />);
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with icon', () => {
        const { container } = render(<PrimaryButton primaryButtonIcon={'Icon'} />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
