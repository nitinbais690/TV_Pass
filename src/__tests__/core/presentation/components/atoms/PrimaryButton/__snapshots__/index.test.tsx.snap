// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primaryButton render with icon 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 3,
        "flex": 1,
        "height": 70,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      hasTVPreferredFocus={true}
      isTVSelectable={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <BVLinearGradient
        colors={
          Array [
            4294929710,
            4290124297,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3B4046",
            "borderColor": "#2089dc",
            "borderRadius": 35,
            "borderWidth": 0,
            "elevation": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 8,
          }
        }
      >
        <Icon
          style={
            Object {
              "marginEnd": 14.571428571428571,
            }
          }
        />
      </BVLinearGradient>
    </View>
  </View>
</View>
`;

exports[`primaryButton renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 3,
        "flex": 1,
        "height": 70,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      hasTVPreferredFocus={true}
      isTVSelectable={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <BVLinearGradient
        colors={
          Array [
            4294929710,
            4290124297,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3B4046",
            "borderColor": "#2089dc",
            "borderRadius": 35,
            "borderWidth": 0,
            "elevation": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 8,
          }
        }
      />
    </View>
  </View>
</View>
`;
